server/storage/interface.go

```diff
@@ -1,14 +1,31 @@
 package storage
 
-// KeyStore provides a minimal interface for managing key persistence
-type KeyStore interface {
-	// GetKey returns the algorithm and public key for the given GUN and role.
-	// If the GUN+role don't exist, returns an error.
-	GetKey(gun, role string) (algorithm string, public []byte, err error)
+import (
+	"time"
+
+	"github.com/docker/notary/tuf/data"
+)
 
-	// SetKey sets the algorithm and public key for the given GUN and role if
-	// it doesn't already exist.  Otherwise an error is returned.
-	SetKey(gun, role, algorithm string, public []byte) error
+// KeyStore provides a minimal interface for managing key persistence.  Any
+// or all of these functions can optionally clean up expired pending keys, or
+// no-longer-active non-pending keys.
+type KeyStore interface {
+	// GetLatestKey the most recently created, non-expired key for the given
+	// GUN and role. If no keys exist for the GUN+role, returns an ErrNoKeys.
+	GetLatestKey(gun, role string) (*ManagedPublicKey, error)
```

Comment:
This won't necessarily be the correct signing key right? It'll literally be the most recently created one?